app/settings/data-export/data-export.controller.js

```diff
@@ -0,0 +1,88 @@
+module.exports = [
+    '$scope',
+    '$rootScope',
+    '$location',
+    'DataExport',
+    'Notify',
+    'FormEndpoint',
+    'FormAttributeEndpoint',
+    '$q',
+    '_',
+function (
+    $scope,
+    $rootScope,
+    $location,
+    DataExport,
+    Notify,
+    FormEndpoint,
+    FormAttributeEndpoint,
+    $q,
+    _
+) {
+    $scope.exportAll = exportAll;
+    $scope.showFields = false;
+    $scope.showProgress = false;
+    $scope.selectFields = selectFields;
+    $scope.selectedFields = [];
+    $scope.exportSelected = exportSelected;
+    $scope.selectAll = selectAll;
+    // hiding export-in-progress screen when export is complete
+    $rootScope.$on('event:data_export:complete', function () {
+        $scope.showProgress = false;
+    });
+
+    // Redirect to home if not authorized
+    if ($rootScope.hasPermission('Bulk Data Import') === false) {
+        return $location.path('/');
+    }
+    // Change layout class
+    $rootScope.setLayout('layout-c');
+    // Change mode
+    $scope.$emit('event:mode:change', 'settings');
+
+    function exportAll() {
+        DataExport.prepareExport({format: 'csv'});
+        $scope.showProgress = true;
+    }
+
+    function selectFields() {
+        $scope.showFields = !$scope.showFields;
+    }
+
+    function getForms() {
+        FormEndpoint.queryFresh().$promise.then(function (response) {
+            $scope.forms = response;
+            var attributeQueries = [];
+            _.each($scope.forms, function (form) {
+                attributeQueries.push(FormAttributeEndpoint.query({formId: form.id}));
+            });
+            $q.all(attributeQueries).then(function (results) {
```

Comment:
@Angamanga 
Ok my suggestion is as follows:
1. Only load the forms and display them as selectable on the page
2. When a user selects one of the forms, then and only then load the attributes
That will be much faster and also reduce the amount of info on the page